core/tests/puppeteer-acceptance-tests/spec/practice-question-reviewer-tests/review-and-accept-question.spec.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Acceptance Test for question admins to add reviewing
+ * question rights to users.
+ */
+
+const userFactory = require(
+  '../../puppeteer-testing-utilities/user-factory.js');
+const testConstants = require(
+  '../../puppeteer-testing-utilities/test-constants.js');
+
+const DEFAULT_SPEC_TIMEOUT = testConstants.DEFAULT_SPEC_TIMEOUT;
+
+describe('Question Admin', function() {
+  let questionAdmin = null;
+
+  beforeAll(async function() {
+    questionAdmin =
+      await userFactory.createNewQuestionAdmin('questionAdm');
+  }, DEFAULT_SPEC_TIMEOUT);
+
+  it('should be able to provide rights to review and submit questions to user.',
+    async function() {
+      let reviewer = await userFactory.createNewGuestUser(
+        'reviewer', 'review@example.com');
+      await userFactory.closeBrowserForUser(reviewer);
+      await questionAdmin.navigateToContributorDashboardAdminPage();
+
+      await questionAdmin.addReviewQuestionRights('reviewer');
```

Comment:
Please add pre-checks to ensure that the user initially has neither reviewing nor contribution rights

Reply:
Thanks for pointing it out, I added a line to call the function expectUserToNotBeDisplayed now.

Reply:
I think `expectUserToNotBeDisplayed()` requires that `viewContributorReviewQuestionRightsByRole()` be called first, right? Otherwise the user won't be displayed simply because we're not on the page that displays privileged users.

More generally, I think it's confusing to separate the functions that navigate to the page that displays user rights from the functions that check those rights. This separation makes it easy to forget to call the navigation function. Instead, would it be clearer to have a single function that both navigates to the appropriate page and checks permissions?

Reply:
To add to this -- I think the comment in https://github.com/oppia/oppia/pull/19178#discussion_r1409594832 might help address this issue.

Reply:
Yes @U8NWXD, you are right. viewContributorReviewQuestionRightsByRole() should be called after expectUserToNotDisplayed() and I agree that it might be confusing to separate it. I will combine them together like @seanlip's comment.